exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/project/ProjectRecordBatch.java

```diff
@@ -122,6 +122,7 @@ protected void killIncoming(final boolean sendUpstream) {
 
   @Override
   public IterOutcome innerNext() {
+    recordCount = 0;
```

Comment:
That wouldn't have worked--the recordCount fields are declared in the subclasses, not in one of those superclasses. 

(I didn't try refactoring recordCount up into one of those base classes because I was already running into enough downstream problems every time I fixed an upstream increment.)